sampling/sample_creator.py

```diff
@@ -1,20 +1,45 @@
-class SampleCreator():
+from multiprocessing import Process
+import random
+
+
+class SampleCreator(Process):
   """
     Responsible for starting multile reservoir samples across
     the memory hierarchy
+    
+    Design reference: https://stackoverflow.com/questions/17172878/using-pythons-multiprocessing-process-class
   """
 
   def __init__(self, data_store):
-    self.data_store = data_store
-    self.target_location = ""
+    super(P, self).__init__()
+    self.ds = data_store
+    self.md = self.data_store.metadata
 
-  def create_samples(self):
+  def run(self):
     """
       Keep creating samples in the background.
     """
-    # Generate keys using reservoir sampling with num_points in 
-    # the metadata file.
+    while True:
+      reservoir = Sample(self.ds.sample_size)
+
+      i = 0
+      while i < reservoir.maxsize and i < self.md.num_points:
+        point = self.get_next_point()
+        reservoir.items.append(point)
+        i += 1
+        
+      while i >= reservoir.maxsize and i < self.md.num_points:
+        point = self.get_next_point()
```

Comment:
This is wasteful, as the point being fetched might not even be used. A better approach will be to run reservoir sampling on the keys alone (data need not be read initially), and after the chosen keys are decided, do one sequential access to fill in the reservoir.